packages/dds/tree/src/feature-libraries/editable-tree/editableTree.ts

```diff
@@ -0,0 +1,444 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { assert } from "@fluidframework/common-utils";
+import { Value, Anchor } from "../../tree";
+import {
+    IEditableForest, TreeNavigationResult, mapCursorField, ITreeSubscriptionCursor, ITreeSubscriptionCursorState,
+} from "../../forest";
+import { brand } from "../../util";
+import {
+    FieldSchema, rootFieldKey, LocalFieldKey, TreeSchemaIdentifier, TreeSchema, ValueSchema,
+} from "../../schema-stored";
+import { FieldKind, Multiplicity } from "../modular-schema";
+import {
+    AdaptingProxyHandler,
+    adaptWithProxy,
+    getFieldKind, getFieldSchema, getPrimaryField, isPrimitive, isPrimitiveValue, PrimitiveValue,
+} from "./utilities";
+
+/**
+ * A symbol for extracting target from editable-tree proxies.
+ * Useful for debugging and testing, but not part of the public API.
+ */
+export const proxyTargetSymbol: unique symbol = Symbol("editable-tree:proxyTarget");
+
+/**
+ * A symbol to get a function, which returns the type of a node in contexts
+ * where string keys are already in use for fields.
+ */
+export const getTypeSymbol: unique symbol = Symbol("editable-tree:getType()");
+
+ /**
+ * A symbol to get the value of a node in contexts where string keys are already in use for fields.
+ */
+export const valueSymbol: unique symbol = Symbol("editable-tree:value");
+
+/**
+ * A tree which can be traversed and edited.
+ *
+ * When iterating, only visits non-empty fields.
+ * To discover empty fields, inspect the schema using {@link getTypeSymbol}.
+ *
+ * TODO: support editing.
+ * TODO: `extends Iterable<EditableField>`
+ * TODO: use proxies for array fields not just raw arrays (will be needed for laziness and editing).
+ * TODO: provide non-schema impacted APIs for getting fields and nodes without unwrapping
+ * (useful for generic code, and when references to these actual fields and nodes are required,
+ * for example creating anchors and editing).
+ */
+export interface EditableTree {
+    /**
+     * A function to get the type of a node.
+     * If this node is well-formed, it must follow this schema.
+     * @param key - if key is supplied, returns the type of a non-sequence child node (if exists)
+     * @param nameOnly - if true, returns only the type identifier
+     */
+    readonly [getTypeSymbol]: (key?: string, nameOnly?: boolean) => TreeSchema | TreeSchemaIdentifier | undefined;
+
+    /**
+     * Value stored on this node.
+     */
+    readonly [valueSymbol]: Value;
+
+    /**
+     * Stores {@link ProxyTarget}, which implements reading and writing for this node.
+     */
+    readonly [proxyTargetSymbol]: object;
```

Comment:
API Extractor is objecting to a link to a non-exported type (that would be a broken link in the docs).
```suggestion
    /**
     * Stores the target for the proxy which implements reading and writing for this node.
     * The details of this object are implementation details,
     * but the presence of this symbol can be used to separate EditableTrees from other types.
     */
    readonly [proxyTargetSymbol]: object;
```